pkg/miniogw/gateway-storj.go

```diff
@@ -5,27 +5,196 @@ package storj
 
 import (
 	"context"
+	"crypto/sha256"
+	"flag"
+	"fmt"
 	"io"
+	"os"
+	"path"
+	"path/filepath"
+	"strconv"
+	"strings"
 	"time"
 
 	"github.com/minio/cli"
+	"github.com/vivint/infectious"
+	"storj.io/storj/pkg/eestream"
 
-	"github.com/minio/minio/cmd"
+	minio "github.com/minio/minio/cmd"
 	"github.com/minio/minio/pkg/auth"
 	"github.com/minio/minio/pkg/hash"
 )
 
+var (
+	pieceBlockSize = flag.Int("piece_block_size", 4*1024, "block size of pieces")
+	key            = flag.String("key", "a key", "the secret key")
+	rsk            = flag.Int("required", 20, "rs required")
+	rsn            = flag.Int("total", 40, "rs total")
+)
+
+// S3CliAPI contains parameters for accessing the Storj network
+type S3CliAPI struct {
+	totalBuckets int
+	bucketlist   []S3BucketList
+}
+
+//S3BucketList structure
+type S3BucketList struct {
+	bucket   minio.BucketInfo
+	filelist S3FileList
+}
+
+//S3FileList structure
+type S3FileList struct {
+	totalFiles int
+	file       minio.ListObjectsInfo
+}
+
+// gS3Gateway global S3 interface structure
+var gS3CliAPI S3CliAPI
```

Comment:
i'd really love to avoid checking in any global variables here. i know we talked about this before, think that could be within scope for this pull request?

Reply:
Modified as suggested to eliminate global references